server/handler.go

```diff
@@ -75,20 +82,25 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	(*mux.Router)(h.router).ServeHTTP(w, r)
 }
 
-type handler struct {
+type robustHandlerFunc func(w http.ResponseWriter, r *http.Request) error
```

Comment:
I like the name you chose, `robustHandlerFunc`, better than what I've used elsewhere, `handlerWithErrorReturnFunc`. :+1: